ChrootBuild.sh

```diff
@@ -6,7 +6,14 @@
 #####################################
 PROGNAME=$(basename "$0")
 CHROOT="${CHROOT:-/mnt/ec2-root}"
-if [[ $(rpm --quiet -q redhat-release-server)$? -eq 0 ]]
```

Comment:
Wasn't ever really planning to support more than just RHEL and CentOS. That said, if we're going to add in support for more than those two, I'd prefer to use a `case` block than continuing to stack `if`/`elif`/`else` block. Most Linux distros include `/etc/os-release` files – I know that RHEL, CentOS, Fedora and Amazon Linux all do. I would prefer reformulating this block to something like:

~~~
case $1 in $( rpm -qf /etc/os-release --qf '%{name}' )
   centos-release)
      OSREPOS=(
         os
         base
         updates
         extras
      )
      ;;
   redhat-release-server)
      OSREPOS=(
         rhui-REGION-client-config-server-7
         rhui-REGION-rhel-server-releases
         rhui-REGION-rhel-server-rh-common
         rhui-REGION-rhel-server-optional
         rhui-REGION-rhel-server-extras
      )
      ;;
   <OTHER_RELEASE_RPM_NAMES>)
      […elided…]
   *)
      echo "Unknown OS. Aborting" >&2
      exit 1
      ;;
esac
~~~

Where your `OSREPOS=(…` content would go into the `<OTHER_RELEASE_RPM_NAMES>)` section of the above.

Reply:
Suggestions sent via sub-PR to @Rendanic